### PR TITLE
[IMP] *: rename incoterm to incoterms

### PR DESCRIPTION
**Pull request by kinhosz**
According to international standards, the abbreviation of 'International Commercial Terms' is INCOTERMS. However, the use of INCOTERM is prevalent in our codebase. This commit fixes it, rewording all references to the correct one. This commit also updates the name of the fields in the models to avoid future confusion about the correct abbreviation.

task: 3995145

ENT-PR: https://github.com/odoo/enterprise/pull/74812
UPG-PR: https://github.com/odoo/upgrade/pull/6847

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -14,10 +14,10 @@
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    incoterm = fields.Many2one(
-        'account.incoterms', 'Incoterm',
+    incoterms = fields.Many2one(
```
Comment (KangOl) [inline addons/sale_stock/models/sale_order.py:17]: Take the opportunity to use the correct pattern for m2o fields.
```suggestion
    incoterms_id = fields.Many2one(
```

File: addons/account/i18n/account.pot [hunk 2]
```diff
@@ -1195,7 +1195,7 @@ msgstr ""
 
 #. module: account
 #: model_terms:ir.ui.view,arch_db:account.report_invoice_document
-msgid "<strong>Incoterm</strong>"
+msgid "<strong>Incoterms</strong>"
```
Comment (ticodoo) [inline addons/account/i18n/account.pot:1198]: never touch the pot file in master, it's an unnecessary diff

**Review by ticodoo — COMMENTED** [review]
This change seems like a bad idea in general imo. From what I can read online:

Incoterm = the specific term you are using
Incoterms = the collection of terms you can use

A lot of the field name/help changes proposed here would imply that more than 1 incoterm can be selected when it isn't the case. I wrote feedback to PO within the task about it

**Comment by kinhosz** [comment]
Task Cancelled
